### PR TITLE
[WNMGDS-3027] Add Web Component Sections to doc site pages.

### PR DESCRIPTION
**Pull request by tamara-corbalt**
## Summary

This pull request is part of a series aimed at adding missing "Web Component" sections to component pages on the documentation site. By including these sections, we ensure developers have quick access to code examples and guidance for the web component version of each component.
- Added "Web Component" sections to the following component pages: Autocomplete, Checkbox, Date Field, Drawer, and Filter Chip.
- Renamed `ReactDocLink` and `ReactDocLinks` components to `StorybookDocLink` and `StorybookDocLinks`, respectively.  
  - Added logic to handle both React and Web Component cases when there are multiple links, such as links to Choice and ChoiceList.
- Corrected a broken link on the Drawer component page.

## How to test
Verify the Storybook links on the following pages to ensure they direct to the correct documentation for both React and Web Component versions:

### Pages with Added Web Component Sections
- [Autocomplete](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/autocomplete/?theme=core#code)
- [Checkbox](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/checkbox/?theme=core#code)
- [Date Field](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/date-field/single-input-date-field/?theme=core#code)
- [Drawer](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/drawer/?theme=core#code)
- [Filter Chip](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/filter-chip/?theme=core#code)
- [Tooltip](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/tooltip/?theme=core#code)
- [Tabs](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/tabs/?theme=core#code)

### Additional Pages with Updated Link Components
Since changes were made to the link components, please also verify the Storybook links on these pages to ensure consistent functionality:
- [Radio](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/radio/?theme=core#code)
- [Step List](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/step-list/?theme=core#code)
- [Table](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/table/?theme=core#code)
- [Vertical Navigation](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/vertical-navigation/?theme=core#code)

Jira story: [WNMGDS-3027](https://jira.cms.gov/browse/WNMGDS-3027)

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.
- [x] Selected appropriate release milestone

### CODE REVIEW

File: packages/docs/content/components/date-field/single-input-date-field.mdx [hunk 1]
```diff
@@ -26,6 +26,10 @@ medicare:
 
 <SeeStorybookForGuidance storyId={'components-singleinputdatefield--docs'} />
 
+### Web Component
```
Comment (tamara-corbalt) [inline packages/docs/content/components/date-field/single-input-date-field.mdx:29]: Should we add a note here to indicate that, although the `ds-date-field` web component and the `SingleInputDateField` React component have different names, they serve as counterparts in functionality?

File: packages/docs/content/components/step-list.mdx
[the remark this answers is not in the document]
Comment (tamara-corbalt) [reply packages/docs/content/components/step-list.mdx:1]: I tested the [link you referenced](https://github.com/CMSgov/design-system/pull/3308/files#diff-a4c74f7793bc17966fcc5cc1f3b992bf6b18392b00d2b9b46efb4b49731a976dR22) at line 22, and it doesn’t appear to be broken in my local environment, on the deployed branch, or in prod. Are we looking at the same link, or could there be another one you’re noticing?

[the remark this answers is not in the document]
Comment (jack-ryan-nava-pbc) [reply packages/docs/content/components/step-list.mdx:1]: Sorry I should be more explicit. The link works, but the content within Storybook is busted, so maybe this is a totally different ticket:

Screenshot 2024-11-08 at 10 49 22 AM

[the remark this answers is not in the document]
Comment (tamara-corbalt) [reply packages/docs/content/components/step-list.mdx:1]: Ohhhhh, you're referring to [the link in the metadata](https://github.com/CMSgov/design-system/pull/3308/files#diff-a4c74f7793bc17966fcc5cc1f3b992bf6b18392b00d2b9b46efb4b49731a976dL9) at line 9. Yes that looks broken and I agree we should fix it! I'll make the change.

**Review by jack-ryan-nava-pbc — APPROVED** [review]
LGTM!

**Review by kim-cmsds — APPROVED** [review]

**Pull request merged**